docs/static/keystore.asciidoc

```diff
@@ -0,0 +1,152 @@
+This means that the environment variable needs to be accessible to the running
+instance of Logstash. This environment variable must also be correctly set for
+any users who need to issue keystore commands (add, list, remove, etc.).
+
+Using a keystore password is optional. The data will be encrypted even if you
+do not set a password. However, it is highly recommended to configure the
+keystore password and grant restrictive permissions to any files that may
+contain the environment variable value.
+
+For example:
+[source,sh]
+--------------------------------------------------
+set +o history
+export LOGSTASH_KEYSTORE_PASS=mypassword
+bin/logstash-keystore create
+--------------------------------------------------
+
+This setup requires the user running Logstash to have the environment variable
+`LOGSTASH_KEYSTORE_PASS=mypassword` defined. If it is not defined, Logstash
+cannot access the the keystore.
+
+When you run Logstash from an RPM or DEB package installation, the environment
+variables are sourced from `/etc/sysconfig/logstash`.
+
+NOTE: You might need to create `/etc/sysconfig/logstash`. This file should be
+owned by `root` with `600` permissions. The expected format of
+`/etc/sysconfig/logstash` is `ENVIRONMENT_VARIABLE=VALUE`, with one entry per
+line.
+
+For other distributions, such as Docker or ZIP, see the documentation for your
+runtime environment (Windows, Docker, etc) to learn how to set the
+environment variable for the user that runs Logstash. Ensure that the
+environment variable (and thus the password) is only accessible to that user.
+
+[float]
+[[keystore-location]]
+=== Keystore location
+
+The keystore must be located in Logstash's `path.settings` directory. This is
+the same directory that contains the `logstash.yml` file. When performing any
+operation against the keystore, it is recommended to set `path.settings` for the
+keystore command. For example, to create a keystore on a RPM/DEB installation:
+
+["source","sh",subs="attributes"]
+----------------------------------------------------------------
+set +o history
+export LOGSTASH_KEYSTORE_PASS=mypassword
+sudo -E /usr/share/logstash/bin/logstash-keystore --path.settings /etc/logstash create
+----------------------------------------------------------------
+
+See <<dir-layout>> for more about the default directory locations.
+
+NOTE: You will see a warning if the `path.settings` is not pointed to the same directory
+as the `logstash.yml`.
+
+[float]
+[[creating-keystore]]
+=== Create a keystore
+
+To create a secrets keystore, use the `create`:
+
+["source","sh",subs="attributes"]
+----------------------------------------------------------------
+bin/logstash-keystore create
+----------------------------------------------------------------
+
+Creates the keystore in the directory defined by the `path.settings`
+configuration setting.
+
+NOTE: It is recommended that you set a <<keystore-password,keystore password>>
```

Comment:
s/before/when/ -- The use of 'before' to me implies that you can add the password after, which isn't true.

Reply:
Done: https://github.com/elastic/logstash/pull/9044/commits/0cf2807f934a20fb338cbca5aa55ffd5782e5e8e